### ISSUE TITLE
strange behavior on windows

### ISSUE THREAD

**Issue by zzanp** [jline/jline3#823]
terminal: alacritty 0.11.0, powershell and cmd.exe
project: [palaiologos/kamilalisp](github.com/palaiologos/kamilalisp)

when i use `car` or `cdr`, a command appears next to the result
this even happens in dumb mode
![Image](https://user-images.githubusercontent.com/42290262/218112182-8a8f438e-e851-42fc-9ed6-eded8a147b5c.png)

**Comment by mattirn** [comment]
@zzanp this is not JLine problem. You should open ticket to project [alaiologos/kamilalisp](https://github.com/jline/jline3/issues/github.com/palaiologos/kamilalisp) instead.

**Comment by kspalaiologos** [comment]
@mattirn Project maintainer here.
The stuff repeated by JLine after the input doesn't contain a newline. There is only one occurence of `System.out.print()` in the KamilaLisp codebase and it's unreachable from the author's example. The issue can also be easily reproduced in IntelliJ terminal:
![obraz](https://user-images.githubusercontent.com/27734421/220202634-72d0468b-c515-4f24-be3a-a6994f13b336.png)
I have hijacked the standard output as follows:
```java
        PrintStream orig = System.out;
        System.setOut(new PrintStream(new OutputStream() {
            @Override
            public void write(int b) throws IOException {
                orig.println("wrote: " + b);
            }
        }));
```
But the issue persists and JLine keeps displaying the bogus data, meaning that it's a problem on your end:
![obraz](https://user-images.githubusercontent.com/27734421/220202902-feec1fc0-fe71-41e2-a0af-e2c3960e7e5b.png)

**Comment by mattirn** [comment]
@kspalaiologos thanks for the stack! I see that the widget `insert-close-paren` fails to update display correctly. I would like to reproduce the problem. Which OS, terminal/emulator and JLine version are you using?
As a work around you can disable 'blink matching parenthesis' feature:
```
lineReader.setVariable(LineReader.BLINK_MATCHING_PAREN, 0)
```

**Comment by kspalaiologos** [comment]
@mattirn I use newest JLine from Maven Central, Linux and IntelliJ terminal emulator - although it happens on Windows and CMD too as you can see from the original report.

**Comment by gnodet** [comment]
> @mattirn I use newest JLine from Maven Central, Linux and IntelliJ terminal emulator - although it happens on Windows and CMD too as you can see from the original report.

I've re-enabled the blinking parentheses on KamilaLisp current v0.3 branch, but could not recreate the issue on OSX + standard terminal.  I've also tried SSH'ing on a linux VM (so from a MacOS terminal) but can't replicate the issue.  I've double checked that the blinking parenthesis does work.  I've tried on windows but I think the windows support is broken because KamilaLisp is missing a dependency on a supported terminal. I've raised https://github.com/kspalaiologos/kamilalisp/pull/84.
